### PR TITLE
Add option to enable f-strings

### LINKED ISSUE #647: fstring support
Hello,
It appears that fstring support is already implemented in starlark-rust but deactivated in Buck2:

https://github.com/facebook/buck2/blob/main/starlark-rust/starlark_syntax/src/dialect.rs#L89 
https://github.com/facebook/buck2/blob/main/app/buck2_interpreter/src/file_type.rs#L24

Currently, there is no straightforward method to activate this feature without recompiling Buck2.

Proposal:
Add a .buckconfig flag to activate fstring support

```
    pub fn dialect(&self, disable_starlark_types: bool, enable_f_strings: bool) -> Dialect {
        let buck_dialect: Dialect = Dialect {
            enable_def: false,
            enable_lambda: true,
            enable_load: true,
            enable_keyword_only_arguments: false,
            enable_types: DialectTypes::Disable,
            enable_load_reexport: false,
            enable_top_level_stmt: false,
            enable_f_strings: enable_f_strings, # gated feature
            ..Dialect::Standard
        };
```

I'm willing to contribute a similar change to buck2, I just don't know where I can add a configuration flag for `.buckconfig`

Thanks,
Hugues

### PR DESCRIPTION
**Pull request by izissise**
This adds `starlark_enable_f_strings` property in `buck2` section. We want to keep it behind a flag for now because some internal tool at facebook doesn't support f-strings yet.
fix #647

### CODE REVIEW

**Comment by izissise** [comment]
@stepancheg Here the PR, I can't assign it to you myself

File: app/buck2_interpreter_for_build/src/interpreter/global_interpreter_state.rs [hunk 1]
```diff
@@ -109,12 +116,23 @@ impl HasGlobalInterpreterState for DiceComputations<'_> {
...
+                let enable_f_strings = ctx
+                    .parse_legacy_config_property(
+                        cell_resolver.root_cell(),
+                        BuckconfigKeyRef {
+                            section: "buck2",
+                            property: "starlark_enable_f_strings",
+                        },
+                    )
+                    .await?
+                    .unwrap_or_else(|| false);
```
[the remark this answers is not in the document]
Comment (JakobDegen) [reply app/buck2_interpreter_for_build/src/interpreter/global_interpreter_state.rs]: Certainly not for `Standard`, given that that's supposed to follow the starlark standard. For `Extended`, it probably should be, but that's starlark so it'd at least need to be a different PR

**Comment by JakobDegen** [comment]
(the import logic is currently having some issues on our side, I'll get this merged as soon as that's unblocked)

File: app/buck2_interpreter_for_build/src/interpreter/global_interpreter_state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (izissise) [reply app/buck2_interpreter_for_build/src/interpreter/global_interpreter_state.rs]: Understood, thanks you for the review

**Pull request closed without merging**